bundle/phases/initialize.go

```diff
@@ -28,6 +29,12 @@ func Initialize() bundle.Mutator {
 			mutator.ExpandWorkspaceRoot(),
 			mutator.DefineDefaultWorkspacePaths(),
 			mutator.SetVariables(),
+			mutator.ResolveVariableReferencesFor(
+				dyn.NewPattern(dyn.Key("variables"), dyn.AnyKey(), dyn.Key("lookup")),
```

Comment:
Is there some kind of rule we can apply where we provide another pattern of paths to not allow resolution for? The logic is now embedded in the main lookup function but it is really specific to this instance.